### PR TITLE
backend first

### PR DESCRIPTION
**Pull request by Abhishek-Geekywolf**

### CODE REVIEW

File: Backend/WolfDen.Infrastructure/Configuration/EmployeeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using WolfDen.Domain.Entity;
+
+namespace WolfDen.Infrastructure.Configuration
+{
+    public class EmployeeConfiguration : IEntityTypeConfiguration<Employee>
+    {
+        
+            public void Configure(EntityTypeBuilder<Employee> builder)
+            {
+                builder.Property(x => x.Id).ValueGeneratedOnAdd().HasColumnName("Id");
```
Comment (sanalsunnyGW) [inline Backend/WolfDen.Infrastructure/Configuration/EmployeeConfiguration.cs:17]: If the property name is same we don't need to specify column names

File: Backend/WolfDen.Infrastructure/Data/WolfDenContext.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        public WolfDenContext(DbContextOptions<WolfDenContext> options):base(options) { }
+        
+            
+        
+        public DbSet<Employee> Employees { get; set; }
+        public DbSet<Department> Departments { get; set; }
+        public DbSet<Designation> Designations { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
```
Comment (sanalsunnyGW) [inline Backend/WolfDen.Infrastructure/Data/WolfDenContext.cs:23]: Try to use a generic method to apply configurations in next PR

File: Backend/WolfDen.API/Program.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.Services.AddDbContext<WolfDenContext>(x =>
+{
+    x.UseSqlServer(@"Server=localhost,1430;Database=EmployeeManagement;User Id=sa;Password=abhi@123;TrustServerCertificate=true");
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.API/Program.cs:15]: move this connection string to appsettings

File: Backend/WolfDen.API/WolfDen.API.http [hunk 4]
```diff
@@ -0,0 +1,6 @@
+@WolfDen.API_HostAddress = http://localhost:5098
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.API/WolfDen.API.http:1]: I think we can remove this file

File: Backend/WolfDen.Domain/Entity/Department.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WolfDen.Domain.Entity
+{
+    public class Department
+    {
+        public int Id { get; private set; }
+        public string DepartmentName { get; private set; }
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.Domain/Entity/Department.cs:12]: DepartmentName to Name

File: Backend/WolfDen.Domain/Entity/Department.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+        public Department()
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.Domain/Entity/Department.cs:14]: make default construtor private

File: Backend/WolfDen.Domain/Entity/Designation.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WolfDen.Domain.Entity
+{
+    public class Designation
+    {
+        public int Id { get; private set; }
+        public string DesignationName { get; private set; }
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.Domain/Entity/Designation.cs:12]: DesignationName  -> Name

File: Backend/WolfDen.Domain/Entity/Employee.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WolfDen.Domain.Entity
+{
+    public class Employee
+    {
+        public int Id { get; private set; }
+        public int EmployeeId { get; private set; }
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.Domain/Entity/Employee.cs:13]: Id and EmployeeId?

File: Backend/WolfDen.Domain/Entity/Employee.cs [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        public string Email { get; private set; }
+        public string PhoneNumber { get; private set; }
+
+        public DateOnly? JoiningDate { get; private set; }
+        public int DesignationId { get; private set; }
+        public Designation Designation { get; private set; }
+        public int DepartmentId {  get; private set; }
+        public Department Department { get; private set; }
+        public int? ManagerId { get; private set; }
+        public virtual Employee Manager {  get; private set; }
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.Domain/Entity/Employee.cs:28]: this is nullable right?

File: Backend/WolfDen.Domain/Entity/Employee.cs [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+
+        public Employee()
```
Comment (anilkumar-geeky) [inline Backend/WolfDen.Domain/Entity/Employee.cs:31]: private

File: Backend/WolfDen.Domain/Entity/Employee.cs (hunk 8 above)
Comment (Abhishek-Geekywolf) [reply Backend/WolfDen.Domain/Entity/Employee.cs:13]: id table id primary key and EmployeeId is Unique id

Comment (sanalsunnyGW) [reply Backend/WolfDen.Domain/Entity/Employee.cs:13]: Use Identifier instead of Id

**Pull request merged**